File: src/dataset/python/anagram_exercism-ThomasZumsteg/claude/ClaudeSonnet4_anagram_v1.py
"""Finds anagrams of a word in a list of words"""

def detect_anagrams(anagram, word_list):
    """Finds any words in [word_list] that are anagrams of [anagram]"""
    anagram_lower = anagram.lower()
    anagram_sorted = sorted(anagram_lower)
    
    result = []
    for word in word_list:
        word_lower = word.lower()
        if word_lower != anagram_lower and sorted(word_lower) == anagram_sorted:
            result.append(word)
    
    return result

File: src/dataset/Java/anagram_exercism-ThomasZumsteg/claude/ClaudeSonnet4_Anagram_v1.java
import java.util.ArrayList;
import java.util.List;
import java.util.Arrays;

public class Anagram {
    private final String anagram;
    private final String anagramLower;
    private final char[] anagramSorted;

    public Anagram(String anagram) {
        this.anagram = anagram;
        this.anagramLower = anagram.toLowerCase();
        this.anagramSorted = this.anagramLower.toCharArray();
        Arrays.sort(this.anagramSorted);
    }

    public List<String> match(List<String> words) {
        List<String> anagrams = new ArrayList<>(words.size());

        for(String word: words) {
            if (word.length() != anagram.length()) {
                continue;
            }
            
            String wordLower = word.toLowerCase();
            
            if (anagramLower.equals(wordLower)) {
                continue;
            }
            
            char[] wordLetters = wordLower.toCharArray();
            Arrays.sort(wordLetters);

            if(Arrays.equals(anagramSorted, wordLetters)) {
                anagrams.add(word);
            }
        }
        return anagrams;
    }
}